File: leetcode/easy/concatenation-of-array.ts
// https://leetcode.com/problems/concatenation-of-array/

// MINE
/*
function getConcatenation(nums: number[]): number[] {
    // cheating
    // return [...nums, ...nums]
    
    // straightforward
    const n = nums.length
    for(let i=0; i<n; ++i) nums.push(nums[i])
    return nums
};
*/

// ANSWER
function getConcatenation(nums: number[]): number[] {
  const ans = new Array(nums.length * 2)
  for (let i = 0; i < nums.length; ++i) ans[i] = ans[nums.length + i] = nums[i]
  return ans
}

File: leetcode/easy/concatenation-of-array.py
# https://leetcode.com/problems/concatenation-of-array/

class Solution:
    # MINE
    def getConcatenation(self, nums: List[int]) -> List[int]:
        # solution1
        # return [*nums, *nums]

        # solution2
        # ans = [None]*len(nums)*2
        # for i in range(len(nums)):
        #     ans[i] = ans[len(nums)+i] = nums[i]
        # return ans

        # solution3
        n = len(nums)
        for i in range(n):
            nums.append(nums[i])
        return nums
